Automattic/wp-calypso
# Media Modal - Remove special case for single external media image selection.  Disable adding transients.

#### Changes proposed in this Pull Request

* Disable the ability to add a transient media selection from media modal.  The modal "insert" button will be disabled if an image whose upload is currently in progress is part of the selection.
* Removes special case of inserting 1 image from the external media section of the media modal.  Single image selections from an external media source will be required to be copied to the library before inserting the image from the modal.  

The main goal of this PR is to ensure media selections added to a site are always done with the proper url from the media library.  The two changes listed above help ensure that the external source url is not used for media added to a users site.  This ensures persistence of the media throughout the lifetime of the site and fixes an issue where images added from google photos would break after an hour.

**Updated Behavior**
![updated](https://user-images.githubusercontent.com/28742426/71040017-99575280-20f3-11ea-8179-138ba0f48e49.gif)

[More reasoning and discussion can be found on the alternative PR](https://github.com/Automattic/wp-calypso/pull/38491).  PR #38491 removes the single image special case for google-photos only, while this PR removes the special case for all external media sources.  After further discussion and thought it seems that removing this special case for all external sources may be the best action.  

#### Testing instructions



Note - it is important to ensure this change works well on Atomic / Jetpack sites and does not negatively effect the user experience.  In reading some past reasoning it seems these uploads may take significantly longer on these sites.

* Run this PR
* Navigate to edit a page.
* Add a gallery block and select the Media Library.
* Try to add images from an external media source (google photos and/or free image library).
* Verify having 1 image selected will copy the item to the media library instead of inserting it.
* Verify you are unable to "Insert" while the image is still selected and uploading into the library.
* Verify inserted images URLs correspond to the media library and not google images.
* Repeat the above for an Atomic or Jetpack site. 

Fixes #34593
